### PR TITLE
Disable proxy button when proxy is added

### LINKED ISSUE #1484: Disable Add proxy button if one proxy exists
While we are currently only supporting one proxy (one API Umbrella per deployment), disable the Add proxy button from Settings / Proxies if administrator has added one proxy to the deployment.

# Definition of done
* [x] Admin users can see the Add proxy button only if no proxy has been added to the platform
* [x] Collection allow/deny rule has been added
  * [x] `insert` is only allowed when no Proxies exist

### PR DESCRIPTION
**Pull request by frenchbread**
Closes #1484

### CODE REVIEW

**Comment by frenchbread** [comment]
TODO: add help text to disabled button

**Comment by bajiat** [comment]
@frenchbread Why not hide the button instead of disabling?

**Comment by frenchbread** [comment]
@bajiat IMHO Hiding the "Add proxy" can confuse users. I have no idea where, in that case, to put a message saying e.g. "You have already added one proxy. Can't add more for now."

Disabling the button was also suggested by @Nazarah [here](https://github.com/apinf/platform/issues/1484#issuecomment-246261613).

What's left is attaching a tooltip with helper text to a button. Open to suggestions.

File: proxies/client/proxies.js [hunk 1]
```diff
@@ -20,12 +20,15 @@ Template.proxies.onCreated(function(){
...
+  },
 });
 
 Template.proxies.helpers({
   proxies () {
     const instance = Template.instance();
     return instance.proxies.get();
-  }
+  },
+  biggerThan (a, b) {
```
Comment (nnn) [inline proxies/client/proxies.js]: Please use more meaningful name variables here, didn't quite get what is a and what is b.
Also comment will help.

File: core/lib/i18n/en.i18n.json [hunk 2]
```diff
@@ -273,6 +273,7 @@
   "proxies_pageHeader": "Proxies",
   "proxies_addProxy": "Add proxy",
   "proxies_noProxiesFound": "No proxies found.",
+  "proxies_cannotAddMoreProxies": "You already have a Proxy available to attach your API. Please select the Proxy. You can add a new Proxy only after your have deleted the existing one.",
```
Comment (brylie) [inline core/lib/i18n/en.i18n.json]: This help text is unnecessary, if we simply **hide the Add Proxy button**.

File: proxies/client/proxies.js [hunk 3]
```diff
@@ -2,30 +2,41 @@ import { Proxies } from '../collection';
...
+
+      // Set button disabled if at least one proxy is already added
+      instance.isDisabled.set(proxiesCount >= 1);
     }
   });
 });
 
 Template.proxies.events({
   'click #add-proxy': function (event, instance) {
     Modal.show('addProxy', { proxy: {}, isEdit: false });
```
Comment (brylie) [inline proxies/client/proxies.js]: Be aware of changes to the `addProxy` form. Specifically, the template has been renamed `proxyForm` and used for both adding and editing proxies.

https://github.com/apinf/platform/pull/1597

File: proxies/collection/permissions.js [hunk 4]
```diff
@@ -2,6 +2,11 @@ import { Proxies } from './';
 
 Proxies.allow({
   insert (userId, proxy) {
+    // Check if proxy is already exist
+    if (Proxies.find().count() >= 1) {
+      return false;
+    }
+
     // Check if user has admin role
     return Roles.userIsInRole(userId, ['admin']);
```
Comment (brylie) [inline proxies/collection/permissions.js]: Add import for  `Roles` package (lint).

**Comment by brylie** [comment]
This is really close. Just clean up the help text, which is not necessary. Also, check linter errors for missing imports, etc.

**Comment by frenchbread** [comment]
Changes are done

**Pull request merged**